In [1]:
pip install tensorflow

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [2]:
# installation des packages
from sklearn import datasets
import numpy as np
import tensorflow as tf
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split

In [3]:
#telecharger la dataset

df=datasets.load_breast_cancer()

In [ ]:
#decouvrir la base de données

print(df.DESCR)

In [ ]:
#variable independates (resultat)
df.target

In [7]:
#les données (caracteristiques)
df.data

array([[1.799e+01, 1.038e+01, 1.228e+02, ..., 2.654e-01, 4.601e-01,
        1.189e-01],
       [2.057e+01, 1.777e+01, 1.329e+02, ..., 1.860e-01, 2.750e-01,
        8.902e-02],
       [1.969e+01, 2.125e+01, 1.300e+02, ..., 2.430e-01, 3.613e-01,
        8.758e-02],
       ...,
       [1.660e+01, 2.808e+01, 1.083e+02, ..., 1.418e-01, 2.218e-01,
        7.820e-02],
       [2.060e+01, 2.933e+01, 1.401e+02, ..., 2.650e-01, 4.087e-01,
        1.240e-01],
       [7.760e+00, 2.454e+01, 4.792e+01, ..., 0.000e+00, 2.871e-01,
        7.039e-02]])

On voit que les données ont des valeurs non standard on va donc les traités avec sklearn.

In [46]:
#preparons les données

X= df.data
Y=df.target

In [47]:
#standarliser les donées (les données seront maintenant entre -1 et 1)

scaler=StandardScaler()
X_scaled = scaler.fit_transform(X)
X_scaled

array([[ 1.09706398, -2.07333501,  1.26993369, ...,  2.29607613,
         2.75062224,  1.93701461],
       [ 1.82982061, -0.35363241,  1.68595471, ...,  1.0870843 ,
        -0.24388967,  0.28118999],
       [ 1.57988811,  0.45618695,  1.56650313, ...,  1.95500035,
         1.152255  ,  0.20139121],
       ...,
       [ 0.70228425,  2.0455738 ,  0.67267578, ...,  0.41406869,
        -1.10454895, -0.31840916],
       [ 1.83834103,  2.33645719,  1.98252415, ...,  2.28998549,
         1.91908301,  2.21963528],
       [-1.80840125,  1.22179204, -1.81438851, ..., -1.74506282,
        -0.04813821, -0.75120669]])

In [48]:
#onva séparer les donnéees en données d'entrainement et en données de test
#on va utiliser 20% des donnéees pour les test donc 0.2
X_train,X_test,Y_train,Y_test=train_test_split(X_scaled,Y,test_size=0.2,random_state=42)



Si vous ne spécifiez pas le random_statedans votre code, chaque fois que vous exécutez (exécutez) votre code, une nouvelle valeur aléatoire est générée et les ensembles de données d'entraînement et de test auraient des valeurs différentes à chaque fois.

Cependant, si une valeur fixe est attribuée comme random_state = 42alors, quel que soit le nombre de fois que vous exécutez votre code, le résultat serait le même, c'est-à-dire les mêmes valeurs dans les ensembles de données d'entraînement et de test.


In [49]:
#tailles de dataset train et test

print(X_train.shape,X_test.shape)

(455, 30) (114, 30)


In [50]:
#construction des réseau de neuronnes
#initialisons notre model(creation du reseau de neuronne)

model=tf.keras.models.Sequential()

In [51]:
#nous allons données le nombre de noeud en entrées c'est à dire le nombre de variables en entrée avec units=nb et aussi l'algorithme d'activation qui est l'algorithme qui permet de determiner la sorties
#les algorithmes d'activation les plus utilisées sont sigmoid, relu,  hyper Tangent et identify

model.add(tf.keras.layers.Dense(units=15,activation='relu',kernel_initializer='uniform',input_dim=30))

#model.add(tf.keras.layers.Dense(units=nb de noeud en entré,activation='algorithme d'activation à utiliser',kernel_initializer='initialisation des poids qui seront corriger au fur avec l'algorithme de cout',input_dim=nombre de variables total de la dataset)


In [52]:
#nous allon maintenant creer la couche de sortie avec l'algorithme sigmoid et non relu cette fois-ci
model.add(tf.keras.layers.Dense(units=1,activation='sigmoid',kernel_initializer='uniform'))


In [53]:
#notre algorithme de cout
#loss est la fonction de cout
model.compile(optimizer='adam',loss='binary_crossentropy',metrics=['binary_accuracy'])

#loss =binary..... parcequ'on a deux sortie donc binaire et metric qui calcul le nombre de fois qu'on se trompe

In [54]:
#visualisons notre reseau de neuronne et voyons les différentes couches ainsi que le nomnre de noeud de chaque couche
model.summary()

Model: "sequential_2"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 dense_4 (Dense)             (None, 15)                465       
                                                                 
 dense_5 (Dense)             (None, 1)                 16        
                                                                 
Total params: 481
Trainable params: 481
Non-trainable params: 0
_________________________________________________________________


In [55]:
#entrainons notre reseau de neuronne(affectation des poids adequats)

model.fit(X_train, Y_train,epochs=100)

#epochs designe le nombre de fois qu'on entraine notre model en ajustant les poids

Epoch 1/100
15/15 [==============================] - 1s 2ms/step - loss: 0.6801 - binary_accuracy: 0.8681
Epoch 2/100
15/15 [==============================] - 0s 2ms/step - loss: 0.6440 - binary_accuracy: 0.9209
Epoch 3/100
15/15 [==============================] - 0s 3ms/step - loss: 0.5859 - binary_accuracy: 0.9275
Epoch 4/100
15/15 [==============================] - 0s 2ms/step - loss: 0.5084 - binary_accuracy: 0.9275
Epoch 5/100
15/15 [==============================] - 0s 2ms/step - loss: 0.4273 - binary_accuracy: 0.9297
Epoch 6/100
15/15 [==============================] - 0s 3ms/step - loss: 0.3520 - binary_accuracy: 0.9363
Epoch 7/100
15/15 [==============================] - 0s 2ms/step - loss: 0.2927 - binary_accuracy: 0.9407
Epoch 8/100
15/15 [==============================] - 0s 2ms/step - loss: 0.2471 - binary_accuracy: 0.9451
Epoch 9/100
15/15 [==============================] - 0s 2ms/step - loss: 0.2151 - binary_accuracy: 0.9516
Epoch 10/100
15/15 [==========================

In [56]:
#testons le model
test_loss,test_accuracy=model.evaluate(X_test,Y_test)
print(f'Test reuissi {test_accuracy} et Test echoué {test_loss}')


4/4 [==============================] - 0s 3ms/step - loss: 0.0512 - binary_accuracy: 0.9912
Test reuissi 0.9912280440330505 et Test echoué 0.05122005194425583


Notre modèle marche très bien avec une précision de 99.12%